frontend/src/pages/blog/Article.js

```diff
@@ -4,9 +4,28 @@ import { useArticlesContext } from "../../hooks/useArticlesContext";
 
 const Article = () => {
   let { id } = useParams();
-  const { articles } = useArticlesContext();
-  const article = articles.filter((a) => a._id === id)[0];
-  return <ArticleView article={article} />;
+  const { articles, dispatch } = useArticlesContext();
+
+  const fetchArticle = async () => {
+      const response = await fetch(`/api/articles/${id}`);  // fetch the article (only this one) from the database
+      if (response.ok) {
+        const article = await response.json();
+        dispatch({ type: "ADD", payload: article });       // if it exists, add it to the ArticlesContext
+        return article;
+      }
+      else {
+        return null;
+      } 
+  };
+  
+  try {
+    const article = articles.filter((a) => a._id === id)[0];
+    return <ArticleView article={article} />;
+  } catch (error) { //if articles are null, articles.filter produces an error
```

Comment:
What kind of error would you like to filter?

Reply:
I don't know the names of errors that can happen but you never now what can trigger an error so better be safe than sorry right? 
In `python` even if I can't *imagine* what kind of Exception could occur I would still filter it as a matter of principle :+1:

Reply:
You are right, but here, regardless the error, we should fetch the article again. 
We don't really need to switch on different type of exception, because in every case, we should fetch.

Reply:
Ok tks for the explanation, I'm ok with it so let's go